src/allotropy/parsers/thermo_fisher_nanodrop_eight/nanodrop_eight_parser.py

```diff
@@ -0,0 +1,271 @@
+from typing import Optional, TypeVar
+import uuid
+
+import pandas as pd
+
+from allotropy.allotrope.models.shared.definitions.custom import (
+    TQuantityValueMicrogramPerMicroliter,
+    TQuantityValueMicrogramPerMilliliter,
+    TQuantityValueMilliAbsorbanceUnit,
+    TQuantityValueMilligramPerMilliliter,
+    TQuantityValueNanogramPerMicroliter,
+    TQuantityValueNanogramPerMilliliter,
+    TQuantityValueNanometer,
+    TQuantityValuePicogramPerMilliliter,
+)
+from allotropy.allotrope.models.shared.definitions.definitions import (
+    TQuantityValue,
+    ValueEnum,
+)
+from allotropy.allotrope.models.spectrophotometry_benchling_2023_12_spectrophotometry import (
+    CalculatedDataAggregateDocument,
+    CalculatedDataDocumentItem,
+    DataSourceAggregateDocument1,
+    DataSourceDocumentItem,
+    DataSystemDocument,
+    DeviceSystemDocument,
+    MeasurementAggregateDocument,
+    Model,
+    SampleDocument,
+    SpectrophotometryAggregateDocument,
+    SpectrophotometryDocumentItem,
+    UltravioletAbsorbancePointDetectionDeviceControlAggregateDocument,
+    UltravioletAbsorbancePointDetectionDeviceControlDocumentItem,
+    UltravioletAbsorbancePointDetectionMeasurementDocumentItems,
+)
+from allotropy.constants import ASM_CONVERTER_NAME, ASM_CONVERTER_VERSION
+from allotropy.named_file_contents import NamedFileContents
+from allotropy.parsers.thermo_fisher_nanodrop_eight.nanodrop_eight_reader import (
+    NanoDropEightReader,
+)
+from allotropy.parsers.vendor_parser import VendorParser
+
+T = TypeVar("T")
+
+CONCENTRATION_UNIT_TO_TQUANTITY = {
+    "ug/ul": TQuantityValueMicrogramPerMicroliter,
+    "ug/ml": TQuantityValueMicrogramPerMilliliter,
+    "mg/ml": TQuantityValueMilligramPerMilliliter,
+    "ng/ul": TQuantityValueNanogramPerMicroliter,
+    "ng/ml": TQuantityValueNanogramPerMilliliter,
+    "pg/ul": TQuantityValuePicogramPerMilliliter,
+}
+
+
+def _get_str(data_frame: pd.DataFrame, row: int, column: str) -> Optional[str]:
+    if column not in data_frame.columns:
+        return None
+    if pd.isna(data_frame.iloc[row][column]):
+        return None
+
+    return str(data_frame.iloc[row][column])
+
+
+def _get_float(data_frame: pd.DataFrame, row: int, column: str) -> float | ValueEnum:
```

Comment:
you can use `JsonFloat` instead of `float | ValueEnum`, here and below. Also note that this is still using Python 3.9 syntax, so if you want a union you need typing.Union.